yt_dlp/extractor/ilpost.py

```diff
@@ -23,46 +23,85 @@ class IlPostIE(InfoExtractor):
-        if not episode:
-            raise ExtractorError('Episode could not be extracted')
+        try:
+            episode = self._search_nextjs_data(webpage, display_id)['props']['pageProps']['data']['data']['episode']['data'][0]
+            episode_id = episode['id']
+        except KeyError:
+            raise ExtractorError('Failed to extract episode')
 
         return {
-            'id': episode_id,
+            'id': str(episode_id),
             'display_id': display_id,
-            'series_id': podcast_id,
-            'vcodec': 'none',
             **traverse_obj(episode, {
-                'title': ('title', {str}),
-                'description': ('description', {str}),
-                'url': ('podcast_raw_url', {url_or_none}),
+                'series_id': ('parent', 'id', {int_or_none}),
+                'title': ('title', {unescapeHTML}),
+                'description': ('content_html', {clean_html}),
+                'url': ('episode_raw_url', {url_or_none}),
                 'thumbnail': ('image', {url_or_none}),
                 'timestamp': ('timestamp', {int_or_none}),
                 'duration': ('milliseconds', {float_or_none(scale=1000)}),
-                'availability': ('free', {lambda v: 'public' if v else 'subscriber_only'}),
+                'availability': ('access_level', {lambda v: 'public' if v == 'all' else 'subscriber_only'}),
             }),
         }
+
+
+class IlPostPodcastIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?ilpost\.it/podcasts/(?P<id>[\w\-]+)[/?#]?$'
+    _TESTS = [{
+        'url': 'https://www.ilpost.it/podcasts/morning/',
+        'info_dict': {
+            'id': 'morning',
+            'display_id': 'morning',
+            'title': 'Morning',
+            'series': 'Morning',
+        },
+        'playlist_mincount': 20,
+    }, {
+        'url': 'https://www.ilpost.it/podcasts/basaglia-e-i-suoi/',
+        'info_dict': {
+            'id': 'basaglia-e-i-suoi',
+            'display_id': 'basaglia-e-i-suoi',
+            'title': 'Basaglia e i suoi',
+            'series': 'Basaglia e i suoi',
+        },
+        'playlist_mincount': 5,
+    }]
+
+    def _real_extract(self, url):
+        display_id = self._match_valid_url(url).group('id')
+        data = self._download_json(f'https://api-prod.ilpost.it/podcast/v1/podcast/{display_id}?hits=20', display_id)
+
+        try:
+            podcast = data['data'][0]['parent']
+        except KeyError:
+            raise ExtractorError('Failed to extract series')
+
+        entries = [{
+            '_type': 'url',
+            'ie_key': 'IlPost',
+            'url': episode['url'],
+            'episode_id': episode['id'],
+            'title': unescapeHTML(episode.get('title')),
+            'description': clean_html(episode.get('content_html')),
+            'series': unescapeHTML(podcast.get('title')),
+        } for episode in traverse_obj(data, ('data'))]
```

Comment:
I'm assuming these episode IDs are like the episode extractor's and need to be stringified? The metadata we yield here also needs to be an exact match of what the episode extractor returns so that `--match-filter` works properly (however it's okay if we return *fewer* fields than the episode extractor)
```suggestion
        entries = [{
            '_type': 'url',
            'ie_key': IlPostIE.ie_key(),
            'url': episode['url'],
            **traverse_obj(episode, {
                'id': ('id', {int_or_none}, {str}),
                'title': ('title', {clean_html}),
                'description': ('content_html', {clean_html}),
                'series': ('parent', 'title', {clean_html}),
            }),
        } for episode in traverse_obj(data, lambda _, v: url_or_none(v['url']))]
```